bios/stage-2/src/main.rs

```diff
@@ -151,17 +173,21 @@ fn start(disk_number: u16, partition_table_start: *const u8) -> ! {
     }
 }
 
-fn load_file(
+fn try_load_file(
     file_name: &str,
     dst: *mut u8,
     fs: &mut fat::FileSystem<disk::DiskAccess>,
     disk: &mut disk::DiskAccess,
     disk_buffer: &mut AlignedArrayBuffer<16384>,
-) -> u64 {
+) -> Option<u64> {
     let disk_buffer_size = disk_buffer.buffer.len();
-    let file = fs
-        .find_file_in_root_dir(file_name, disk_buffer)
-        .expect("file not found");
+    let optional_file = fs.find_file_in_root_dir(file_name, disk_buffer);
+    if optional_file.is_none() {
+        return None;
+    }
+
+    let file = optional_file.unwrap();
```

Comment:
We can use a `match` here to avoid the `unwrap`:

```suggestion
    let optional_file = fs.find_file_in_root_dir(file_name, disk_buffer);
    let file = match optional_file {
        Some(file) => file,
        None => return None,
    };
```

Alternatively, we might even be able to use the question mark operator here:
```suggestion
    let file = fs.find_file_in_root_dir(file_name, disk_buffer)?;
```

(not sure if this works, I haven't tested it)